src/pybind/mgr/devicehealth/module.py

```diff
@@ -0,0 +1,257 @@
+
+"""
+Device health monitoring
+"""
+
+import errno
+import json
+from mgr_module import MgrModule, CommandResult
+import rados
+from threading import Event
+from datetime import datetime, timedelta, date, time
+
+TIME_FORMAT = '%Y%m%d-%H%M%S'
+
+class Module(MgrModule):
```

Comment:
Would be good to have a self test mode (plus call from qa/tasks/mgr/test_module_selftest) that picks an OSD ID, runs scrape on it, then reads back the result.